apple-codesign/src/macho_signing.rs

```diff
@@ -629,15 +629,15 @@ impl<'data> MachOSigner<'data> {
             }
         }
 
-        if !requirements.is_empty() {
-            info!("code requirements: {}", requirements);
+        info!("code requirements: {}", requirements);
+        let mut blob = RequirementSetBlob::default();
 
-            let mut blob = RequirementSetBlob::default();
+        if !requirements.is_empty() {
             requirements.add_to_requirement_set(&mut blob, RequirementType::Designated)?;
-
-            res.push((CodeSigningSlot::RequirementSet, blob.into()));
         }
 
+        res.push((CodeSigningSlot::RequirementSet, blob.into()));
```

Comment:
Does this line need to be moved? I think a side-effect of your changes in `parse_data()` is that we'll take the `if !requirements.is_empty()` branch now and will add the blob where we weren't before.

I do like always logging a message regarding code requirements: that can be useful for debugging. But perhaps we should log an explicit message in the empty case instead of just dumping the string. After all, the string can be empty, right?

Reply:
> Does this line need to be moved? I think a side-effect of your changes in parse_data() is that we'll take the if !requirements.is_empty() branch now and will add the blob where we weren't before.

Without the requirement set present codesign error with "the sealed resource directory is invalid", that's why I moved the code around.

> I do like always logging a message regarding code requirements: that can be useful for debugging. But perhaps we should log an explicit message in the empty case instead of just dumping the string. After all, the string can be empty, right?

I addressed this by having the same behaviur as before (!requirements.is_empty() => print)

Reply:
OK. If we need the requirement set to avoid the `the sealed resource directory is invalid` error, then we should capture that important context in a code comment. Without a comment warning about the nuance, I could easily see someone changing the logic because emitting an empty requirement set feels odd!